source/main.cpp

```diff
@@ -251,6 +255,16 @@ void GameLoop(PlayerInfo &player, Conversation &conversation, bool &debugMode)
 		// Tell all the panels to step forward, then draw them.
 		((!isPaused && menuPanels.IsEmpty()) ? gamePanels : menuPanels).StepAll();
 		
+		// Currently running the hardcoded "empty" testcase;
+		//   Wait for the game to be fully loaded and then quit.
+		//
+		// This testcase can catch issues related to startup/data-loading,
+		// and it helps to show if the CI testframework is working.
+		// This hardcoded testcase is expected to be replaced by the
+		// larger testframework that is planned for ES.
+		if(!testToRun.empty() && GameData::Progress() == 1)
```

Comment:
[GameData::Progress](https://github.com/endless-sky/endless-sky/blob/b274eae897e2aa4c70f187f8f016dd56a47da73a/source/GameData.cpp#L312) is not a simple accessor, unfortunately, despite its name implying it is.

Reply:
Would it be okay if I move the isReady variable from https://github.com/endless-sky/endless-sky/blob/f3edfc4cd4cfdfcf37b1f90b81d511cdd555868a/source/MenuPanel.cpp#L46 to the UI class, set (and check) it from MenuPanel and also check it from here?

Reply:
That sounds reasonable for now. `IsReady` might be a little ambiguous of a name, though.

(Ideally, `GameData` would be able to tell you that it had finished loading the game data it was asked to load)